src/DotVVM.Framework/Controls/DotvvmControl.cs

```diff
@@ -283,6 +298,7 @@ protected virtual void RenderControl(IHtmlWriter writer, IDotvvmRequestContext c
             this.properties.TryGet(DataContextProperty, out var dc);
             r.DataContext = dc as IValueBinding;
             r.HasActives = true;
+            r.HasActiveGroups = true;
```

Comment:
I have changed the code to use the `TouchProperty` instead of pessimistically assuming that it contains active groups.

I have done some benchmarking and now it's on par with master. Before it allocated 400 more bytes per controls and the control rendering was about 10-20% slower